clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -959,17 +959,22 @@ void SYCLToolChain::AddImpliedTargetArgs(const llvm::Triple &Triple,
                                          const JobAction &JA) const {
   // Current implied args are for debug information and disabling of
   // optimizations.  They are passed along to the respective areas as follows:
-  //  FPGA and default device:  -g -cl-opt-disable
-  //  GEN:  -options "-g -O0"
-  //  CPU:  "--bo=-g -cl-opt-disable"
+  // FPGA:  -g -cl-opt-disable
+  // Default device AOT: -g -cl-opt-disable
+  // Default device JIT: -g (-cl-opt-disable is handled by the runtime)
```

Comment:
1. Are you saying that the driver implicitly passes `-g` option by default in SYCL mode?
2. Could you clarify how the runtime identifies when to pass `-cl-opt-disable` option, please?

Reply:
1. If the user passes -g to clang++, we pass -g to clang-offload-wrapper which goes into the image that the sycl runtime extracts and passes to the backend. That already happened and is unrelated to this change

2. The logic is [here](https://github.com/intel/llvm/blob/bda1eb21c179213f004c4c2c41696f2fef180070/sycl/source/detail/program_manager/program_manager.cpp#L453), it was implemented by @asudarsa. That change stores the opt level as a seperate property in the image, unrelated to the ones added by the driver in 1). It seems the intended of this change was to not pass the flag to the backend in some cases (mostly relevant to me, if it's ESIMD), and that logic works, but there was a second instance of the flag coming from the driver, preventing the change from working as expected. This change fixes that.

Reply:
1. Thanks for clarification. I think the original goal of the `AddImpliedTargetArgs` was to add implied -device flags when user passes `-fsycl-targets=intel_gpu_*` to the driver. This comment says about translating -g* and -O* driver options to device compiler options, but it's not clear why it happens in `AddImpliedTargetArgs`. I thought that this method implicitly adds options to the device compiler rather than processes explicitly set driver options.
2. Probably it would be more accurate to say that `-O0` is handled by the runtime rather than `-cl-opt-disable`.

Reply:
1) Yeah I didn't add the original code that hands -O0 and -g here, so I can't comment on that. @mdtoguchi Any ideas?

2) Sure, I will update the comment, good point.